From: "Yan, Zheng" <zyan@redhat.com>
Date: Fri, 13 May 2016 11:30:24 +0800
Subject: ceph: make ceph_update_writeable_page() uninterruptible
Git-commit: a78bbd4b29c29784f0addb5e3b35790c7ed178ae
Patch-mainline: v4.7-rc1
References: FATE#322288

ceph_update_writeable_page() is used by ceph_write_begin(). It beaks
atomicity of write operation if it's interruptible.

Signed-off-by: Yan, Zheng <zyan@redhat.com>
Acked-by: Luis Henriques <lhenriques@suse.com>
---
 fs/ceph/addr.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fs/ceph/addr.c b/fs/ceph/addr.c
index 97ee5d1fbb61..4aa8e375e648 100644
--- a/fs/ceph/addr.c
+++ b/fs/ceph/addr.c
@@ -1168,7 +1168,7 @@ static int ceph_update_writeable_page(struct file *file,
 			snapc = ceph_get_snap_context(snapc);
 			unlock_page(page);
 			ceph_queue_writeback(inode);
-			r = wait_event_interruptible(ci->i_cap_wq,
+			r = wait_event_killable(ci->i_cap_wq,
 			       context_is_writeable_or_written(inode, snapc));
 			ceph_put_snap_context(snapc);
 			if (r == -ERESTARTSYS)

